### PR TITLE
Added versioning support for InMemoryFullArchetypeRepository

### PR DESCRIPTION
**Pull request by boeckers**
Added `ArchetypeHRIDMap` to support versioning in `InMemoryFullArchetypeRepository`.

I am not sure whether the `getIdUpToConcept` method should be added to `ArchetypeHRID` since it deviates from the openEHR specifications.

### CODE REVIEW

File: tools/src/test/java/com/nedap/archie/flattener/InMemoryFullArchetypeRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            inMemoryFullArchetypeRepository.setFlattenedArchetype(archetypes.get(i));
+            inMemoryFullArchetypeRepository.setValidationResult(validationResults.get(i));
+            inMemoryFullArchetypeRepository.setOperationalTemplate(operationalTemplates.get(i));
+        }
+    }
+
+    @Test
+    public void getArchetype() {
+        assertEquals("openEHR-EHR-OBSERVATION.blood_pressure.v2.0.0",
+                inMemoryFullArchetypeRepository.getArchetype("openEHR-EHR-OBSERVATION.blood_pressure.v").getArchetypeId().getFullId());
```
Comment (pieterbos) [inline tools/src/test/java/com/nedap/archie/flattener/InMemoryFullArchetypeRepositoryTest.java]: In the spec I think a version number of at least 1 digit is always required. But maybe good to support a lookup of the latest version of an archetype regardless of a version. In that case, should we include the `.v` or leave it out altogether?

File: tools/src/main/java/com/nedap/archie/flattener/ArchetypeHRIDMap.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public T getLatestVersion(ArchetypeHRID archetypeHRID){
+
+        //Early return for fully defined version
+        if (archetypeHRID.getPatchVersion() != null) {
+            return this.get(archetypeHRID);
+        }
+
+        //Filter if version is not fully defined
+        List<ArchetypeHRID> keys = this.keySet().stream().
+//                filter(id -> (id.getNamespace() == null && archetypeHRID.getNamespace() == null) ||
```
Comment (pieterbos) [inline tools/src/main/java/com/nedap/archie/flattener/ArchetypeHRIDMap.java]: Perhaps remove this comment after we decide to keep the getIdUpToConcept() method :)

Comment (boeckers) [reply tools/src/main/java/com/nedap/archie/flattener/ArchetypeHRIDMap.java]: Removed comment.

File: tools/src/test/java/com/nedap/archie/flattener/InMemoryFullArchetypeRepositoryTest.java (hunk 1 above)
Comment (boeckers) [reply tools/src/test/java/com/nedap/archie/flattener/InMemoryFullArchetypeRepositoryTest.java]: Removed the `.v` for that case. This required some more logic in the `ArchetypeHRID` (and similarly in the `ArchetypeID`). Maybe we should think of adding fields for fullID, semanticID, etc. at some point?

File: openehr-rm/src/main/java/com/nedap/archie/rm/support/identification/ArchetypeID.java [hunk 3]
```diff
@@ -199,10 +209,12 @@ public void setRmEntity(String rmEntity) {
         this.rmEntity = rmEntity;
     }
 
+    @Deprecated
```
Comment (pieterbos) [inline openehr-rm/src/main/java/com/nedap/archie/rm/support/identification/ArchetypeID.java]: Why is this deprecated?  It's not in the OpenEHR spec, as far as I know?

**Pull request merged**